### ISSUE TITLE
expo-template-js crashes

### ISSUE THREAD

**Issue by AntonVanGoethem** [realm/realm-js#5645]
### How frequently does the bug occur?

Always

### Description

**Expected:**
I open the app on my phone and see the template, can start working with it.
**Actual:**
I open the app, select the development server, and the app crashes.

Please let me know if I can supply more information.

### Can you reproduce the bug?

Always

### Reproduction Steps

1. Fetch template:

`expo init MyAwesomeRealmApp --template @realm/expo-template-js`

2. Build using eas:

`eas build --profile development --platform ios`

3. Run dev client:

`npx expo start --dev-client`

### Version

11.3.1

### What services are you using?

Both Atlas Device Sync and Atlas App Services

### Are you using encryption?

No

### Platform OS and version(s)

iOS 16.3.1

### Build environment

built with eas

**Comment by kneth** [comment]
@AntonVanGoethem Please take a look at https://github.com/realm/realm-js/issues/5276 as I believe the question is similar to yours.

**Comment by AntonVanGoethem** [comment]
The issue discussed there was about the android simulator failing while for me it is the development build crashing on iOS. When I build and install an template expo app without realm it works fine, but when I build the realm template using eas and install on my phone the app crashes every time I try to open it.

**Comment by isaacattaiwan** [comment]
This is happening to me on Android (device and emulator) as well. Crashes on startup.

**Comment by jazzywhit** [comment]
I am new to using this template, but I have not been able to run the app as it crashes immediately upon opening. I have followed the instructions and receive no errors or issues before launching the app. I have also tried on expo-template-ts and have the same result.

<img width="328" alt="image" src="https://user-images.githubusercontent.com/1100200/229156993-f9e682f9-7d12-4d55-8022-c51263f9d4c6.png">
